# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: MIT

From bernhard.rosenkraenzer.ext@huawei.com Tue Jun 29 22:01:00 2021
From: Bernhard Rosenkraenzer <bernhard.rosenkraenzer.ext@huawei.com>
Date: Tue, 29 Jun 2021 22:01:00 +0100
Subject: [PATCH] Don't assume libunwind implementation details

lib/isc/backtrace.c currently assumes _Unwind_Backtrace and _Unwind_GetIP are
implemented as functions.
This is not always true -- some versions of libunwind (such as the 32-bit ARM
implementation built into some versions of gcc) implement _Unwind_GetIP as a
macro, resulting in an undefined reference to _Unwind_GetIP instead of
expanding the macro.

Signed-off-by: Bernhard Rosenkraenzer <bernhard.rosenkraenzer.ext@huawei.com>
Upstream-Status: Pending

diff --git a/lib/isc/backtrace.c b/lib/isc/backtrace.c
index 69488de..8da2c61 100644
--- a/lib/isc/backtrace.c
+++ b/lib/isc/backtrace.c
@@ -82,10 +82,7 @@ isc_backtrace_gettrace(void **addrs, int maxaddrs, int *nframes) {
 	return (ISC_R_SUCCESS);
 }
 #elif defined(BACKTRACE_GCC)
-extern int
-_Unwind_Backtrace(void *fn, void *a);
-extern void *
-_Unwind_GetIP(void *ctx);
+#include <unwind.h>
 
 typedef struct {
 	void **result;
